enumerables.rb

```diff
@@ -0,0 +1,186 @@
+
+  def my_all?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        return false unless i.class == Integer || i.class == Float || i.Complex
+      end
+      true
+    elsif default.class == Class
+      elements.my_each do |i|
+        return false unless i.class == default
+      end
+      true
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return false unless i =~ default
+      end
+      true
+    elsif block_given?
+      elements.my_each do |i|
+        return false unless yield i
+      end
+      true
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return false unless i
+      end
+      true
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return false unless i == default
+      end
+      true
+    end
+  end
+
+  def my_any?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        return true if i.class == Integer || i.class == Float || i.Complex
+      end
+      false
+    elsif default.class == Class
+      elements.my_each do |i|
+        return true if i.class == default
+      end
+      false
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return true if i =~ default
+      end
+      false
+    elsif block_given?
+      elements.my_each do |i|
+        return true if yield i
+      end
+      false
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return true if i
+      end
+      false
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return true if i == default
+      end
+      false
+    end
+  end
+
+  def my_none?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        return true unless i.class == Integer || i.class == Float || i.Complex
+      end
+      false
+    elsif default.class == Class
+      elements.my_each do |i|
+        return false if i.class == default
+      end
+      true
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return false if i =~ default
+      end
+      true
+    elsif block_given?
+      elements.my_each do |i|
+        return false if yield i
+      end
+      true
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return false if i
+      end
+      true
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return false if i == default
+      end
+      true
+    end
+  end
+
+  def my_count(default = nil)
+    elements = to_a
+    count = 0
+    if default
+      elements.my_each do |i|
+        count += 1 if i == default
+      end
+    elsif block_given?
+      elements.my_each do |i|
+        count += 1 if (yield i) == true
+      end
+    else
+      elements.my_each do
+        count += 1
+      end
+    end
+    count
+  end
```

Comment:
Fix the following for `my_count` method

1). **It should return the length of the array if no argument or block is given.**
```
print [1, 2, 4, 2].count   #=> 4
print [1, 2, 4, 2].my_count   #=> enumerables.rb:44:in `my_count': wrong number of arguments (given 0, expected 1) (ArgumentError)
```
2). **If a block is given, it counts the number of elements yielding a true value.**
```
print [1, 2, 4, 2].count{ |x| x%2==0 } #=> 3
print [1, 2, 4, 2].my_count{ |x| x%2==0 }  #=>enumerables.rb:44:in `my_count': wrong number of arguments (given 0, expected 1) (ArgumentError)
```
3). **It should iterate over a range of values.**
```
print (1..5).count{ |x| x%2==0 } #=> 2
print (1..5).my_count{ |x| x%2==0 }  #=> enumerables.rb:44:in `my_count': wrong number of arguments (given 0, expected 1) (ArgumentError)
```